docs/admin/master-node-communication.md

```diff
@@ -0,0 +1,109 @@
+
+**Table of Contents**
+<!-- BEGIN MUNGE: GENERATED_TOC -->
+
+- [Master <-> Node Communication](#master---node-communication)
+  - [Summary](#summary)
+  - [Cluster -> Master](#cluster---master)
+  - [Master -> Cluster](#master---cluster)
+    - [SSH Tunnels](#ssh-tunnels)
+
+<!-- END MUNGE: GENERATED_TOC -->
+
+## Summary
+
+This document catalogs the communication paths between the master (really the
+apiserver) and the Kubernetes cluster. The intent is to allow users to
+customize their installation to harden the network configuration such that
+the cluster can be run on an untrusted network (or on fully public IPs on a
+cloud provider).
+
+## Cluster -> Master
+
+All communication paths from the cluster to the master terminate at the
+apiserver (none of the other master components are designed to expose remote
+services). In a typical deployment, the apiserver is configured to listen for
+remote connections on a secure HTTPS port (443) with one or more forms of
+client [authentication](authentication.md) enabled.
+
+Nodes should be provisioned with the public root certificate for the cluster
+such that they can connect securely to the apiserver along with valid client
+credentials. For example, on a default GCE deployment, the client credentails
+provided to the kubelet are in the form of a client certificate. Pods that
+wish to connect to the apiserver can do so securely by leveraging a service
+account so that Kubernetes will automatically inject the public root
+certificate and a valid bearer token into the pod when it is instantiated.
+The `kubernetes` service (in all namespaces) is configured with a virtual IP
+address that is redirected (via kube-proxy) to the HTTPS endpoint on the
+apiserver.
+
+As a result, the default operating mode for connections from the cluster
+(nodes and pods running on the nodes) to the master is secured by default
+and can run over untrusted and/or public networks.
+
+## Master -> Cluster
+
+There are two primary communication paths from the master (apiserver) to the
+cluster. The first is from the apiserver to the kubelet process which runs on
+each node in the cluster. The second is from the apiserver to any node, pod,
+or service through the apiserver's proxy functionality.
+
+The connections from the apiserver to the kubelet are used for fetching logs
+for pods, attaching (through kubectl) to running pods, and using the kubelet's
+port-forwarding functionality. These connections terminate at the kubelet's
+HTTPS endpoint, which is typically using a self-signed certificate, and
+ignore the certificate presented by the kubelet. These connections are not
+currently safe to run over untrusted and/or public networks as they are
+subject to man-in-the-middle attacks.
+
+The connections from the apiserver to a node, pod, or service default to plain
+HTTP connections and are therefore neither authenticated nor encrypted. These
+connections are not currently safe to run over untrusted and/or public networks.
```

Comment:
That's on a per-request basis though (e.g no policy can be applied that will automatically force all of connections to be secure). I've added this to the doc though since it's useful for people to know about.